cmd/cli/chat.go

```diff
@@ -22,31 +37,234 @@ func addChatCommand() {
 		ValidArgsFunction: cobra.NoFileCompletions,
 		RunE:              chat,
 	}
+	cmd.PersistentFlags().StringVar(&chatModelName, "model", "", "name of the model to use")
+	cmd.PersistentFlags().StringVar(&chatBaseUrl, "base-url", "", "base URL of the OpenAI API server")
 	rootCmd.AddCommand(cmd)
 }
 
 func chat(_ *cobra.Command, args []string) error {
-	// Run the app at path set in CHAT environment variable
-	chatPath, found := os.LookupEnv(envChat)
-	if !found {
-		return fmt.Errorf(`"chat" command not supported`)
+
+	if chatBaseUrl == "" {
+		apiUrls, err := serverApiUrls()
+		if err != nil {
+			return fmt.Errorf("error getting server api urls: %v", err)
+		}
+		chatBaseUrl = apiUrls[openAi]
+	}
+	if verboseLogging {
+		log.Printf("Base URL %v\n", chatBaseUrl)
+	}
+
+	reasoningModel := os.Getenv("REASONING_MODEL") == "true"
+	if verboseLogging && reasoningModel {
+		log.Printf("Reasoning model\n")
+	}
+
+	if chatModelName == "" {
+		var err error
+		chatModelName, err = findModelName()
+		if err != nil {
+			return err
+		}
+	}
+	if verboseLogging {
+		log.Printf("Using model %v\n", chatModelName)
+	}
+
+	// OpenAI API Client
+	client := openai.NewClient(openaiOption.WithBaseURL(chatBaseUrl))
+
+	// TODO: is checkServer needed, as we already communicate with the server when looking up the model name
+	if err := checkServer(chatBaseUrl, client, chatModelName); err != nil {
+		return fmt.Errorf("%v\n\nUnable to chat. Make sure the server has started successfully.\n", err)
```

Comment:
Ideally, error messages should not include line breaks and make assumptions on how the caller uses them.

Try:
```suggestion
        fmt.Println("Unable to chat. Make sure the server has started successfully.\n")
		return err
```